src/tests/functional/shared_test_classes/src/subgraph/matmul_split_decompose.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/subgraph/matmul_split_decompose.hpp"
+
+#include "common_test_utils/graph_comparator.hpp"
+#include "functional_test_utils/skip_tests_config.hpp"
+#include "openvino/pass/manager.hpp"
+#include "openvino/runtime/exec_model_info.hpp"
+#include "transformations/common_optimizations/matmul_split_decomposition.hpp"
+#include "common_test_utils/data_utils.hpp"
+
+namespace ov {
+namespace test {
+
+inline void CheckNumberOfNodesWithType(std::shared_ptr<const ov::Model> function,
+                                       const std::unordered_set<std::string>& nodeTypes,
+                                       size_t expectedCount) {
+    ASSERT_NE(nullptr, function);
+    int num_ops = 0;
+    for (const auto& node : function->get_ordered_ops()) {
+        const auto& rt_info = node->get_rt_info();
+        const auto layer_type = rt_info.find("layerType")->second.as<std::string>();
+        if (nodeTypes.count(layer_type)) {
+            num_ops++;
+        }
+    }
+    ASSERT_EQ(num_ops, expectedCount);
+}
+
+std::string MatMulGatherDecompose::getTestCaseName(const testing::TestParamInfo<MatMulGatherDecomposeParams>& obj) {
+    MatMulGatherDecomposeShapeParams shape_params;
+    std::string device;
+    bool enable_fq;
+    std::tie(shape_params, device, enable_fq) = obj.param;
+    std::ostringstream results;
+
+    results << "input=" << shape_params.input_shape << "_";
+    results << "weights=" << shape_params.weights_shape << "_";
+    results << "trans_b=" << std::boolalpha << shape_params.trans_b << "_";
+    results << "have_bias=" << std::boolalpha << shape_params.have_bias << "_";
+    if (shape_params.have_bias) {
+        results << "bias=" << shape_params.bias_shape << "_";
+    }
+    results << "reshape=" << shape_params.reshape_shape << "_";
+    results << "dev=" << device << "_";
+    results << "enable_fq=" << enable_fq;
+
+    return results.str();
+}
+
+void MatMulGatherDecompose::SetUp() {
+    MatMulGatherDecomposeShapeParams shape_params;
+    element::Type precision = element::f32;
+    bool enable_fq = false;
+    std::tie(shape_params, targetDevice, enable_fq) = GetParam();
+
+    const auto& input_shape = shape_params.input_shape;
+    const auto& weights_shape = shape_params.weights_shape;
+    const auto& have_bias = shape_params.have_bias;
+    const auto& bias_shape = shape_params.bias_shape;
+    const auto& reshape_shape = shape_params.reshape_shape;
+
+    auto param = std::make_shared<ov::op::v0::Parameter>(precision, input_shape);
+
+    std::vector<float> weights_vals(shape_size(weights_shape), 2.0f);
+    weights_vals = ov::test::utils::generate_float_numbers(shape_size(weights_shape), -0.1f, 0.1f);
+    auto weights = ov::op::v0::Constant::create(precision, weights_shape, weights_vals);
```

Comment:
BTW we can use `ov::test::utils::make_constant` to generate constant with random data. The generation range is applied automatically in this function, based on precision

Reply:
Updated